### PR TITLE
Enabled Query Page to Function

### PR DESCRIPTION
**Pull request by Jasminewatt2100**
- Created a query interface with instrument query fields as a drag down tag.
- Created a result page after being queried. 
- Created a display table for query results.

### CODE REVIEW

**Comment by DinoBektesevic** [comment]
Thanks for committing this work, could you fix the flake8 errors before code review starts?

File: trail/query/templates/query.html [hunk 1]
```diff
@@ -1,19 +1,54 @@
...
 <section class="post">
 		<header class="major">
 				<h1>This is where <br />
 						the query lives</h1>
-				<p>Aenean ornare velit lacus varius enim ullamcorper proin aliquam<br />
-						facilisis ante sed etiam magna interdum congue. Lorem ipsum dolor<br />
-						amet nullam sed etiam veroeros.</p>
+				<p>The placeholder is removed here.<br />
+						Here as well.<br />
+						So as here.</p>
```
Comment (mrawls) [inline trail/query/templates/query.html]: Please put some useful text here. The `h1` header could read "Query Tool" and a single paragraph could read, for example, "Query the metadata database generated from previously uploaded FITS files below."

File: trail/query/templates/query.html [hunk 2]
```diff
@@ -1,19 +1,54 @@
...
+</form>
+
+{% if render_table %}
+<h2>Results</h2>
+<table style="width:100%">
+    <tr>
+        <th>Instrument</th>
+        <th>Telescopename</th>
+        <th>datetime_begin</th>
+        <th>exposureduration</th>
```
Comment (mrawls) [inline trail/query/templates/query.html]: Please format these three table headers in a more human-readable friendly way. I suggest "Telescope," "Observation Start," and "Duration (s)".

File: trail/query/views.py [hunk 3]
```diff
@@ -1,6 +1,42 @@
...
+from django.apps import apps
+from django.shortcuts import render
+
+
+Metadata = apps.get_model('upload', 'Metadata')
+# Needs to think about adding SELECT DISTINCT etc.
+
+
+class MetadataForm(forms.Form):
+    unique_instrument = ["DECam", "ff09", "text", "option"]
```
Comment (mrawls) [inline trail/query/views.py]: Please populate this list with actual instrument names that exist in the metadata database. I know for a fact "text" and "option" are invalid 🙂

File: trail/query/models.py [hunk 4]
```diff
@@ -1,3 +1,5 @@
-# from django.db import models
+from django.db import models
 
-# Create your models here.
+
+class Observation(models.Model):
```
Comment (mrawls) [inline trail/query/models.py:4]: Please add a docstring to this class. You can follow the guide at https://developer.lsst.io/python/numpydoc.html#documenting-classes

File: trail/query/views.py [hunk 5]
```diff
@@ -1,6 +1,42 @@
...
+# from django.forms import ModelForm
+from django.apps import apps
+from django.shortcuts import render
+
+
+Metadata = apps.get_model('upload', 'Metadata')
+# Needs to think about adding SELECT DISTINCT etc.
+
+
+class MetadataForm(forms.Form):
```
Comment (mrawls) [inline trail/query/views.py:11]: Please add a docstring to this class. Again, I suggest following the guide at https://developer.lsst.io/python/numpydoc.html#documenting-classes

File: trail/query/views.py [hunk 6]
```diff
@@ -1,6 +1,42 @@
...
+    checking for results are performed.
+
+    It is assumed the request type is GET.
+    """
+    form = MetadataForm()
+    return render(request, "query.html", {"queryform": form, "render_table": False})
+
+
+def print_results(request):
+    """Renders the results url, which is a placeolder copy of the root url of
```
Comment (mrawls) [inline trail/query/views.py]: Typo - "placeholder" not "placeolder"

**Review by mrawls — COMMENTED** [review]
Thank you for your hard work this summer Jasmine! Please address my review comments by following the add+commit+push workflow. I also suggest we work through an interactive rebase together before you merge this branch, so the commit history is a bit cleaner.

File: trail/query/views.py (hunk 3 above)
Comment (DinoBektesevic) [reply trail/query/views.py]: If you can query for unique values from `Metadata` instrument column that would be even better.

File: trail/query/templates/query.html (hunk 1 above)
Comment (Jasminewatt2100) [reply trail/query/templates/query.html]: Fixed,

File: trail/query/templates/query.html (hunk 2 above)
Comment (Jasminewatt2100) [reply trail/query/templates/query.html]: Fixed.

File: trail/query/views.py (hunk 3 above)
Comment (Jasminewatt2100) [reply trail/query/views.py]: Fixed to instrument2 and instrument3, but I need more information about what instruments can be there.

File: trail/query/views.py (hunk 6 above)
Comment (Jasminewatt2100) [reply trail/query/views.py]: This note is not made by me but I have fixed it.

File: trail/query/views.py (hunk 3 above)
Comment (mrawls) [reply trail/query/views.py]: @DinoBektesevic I agree this would be better, but we swapped in a hard-coded value for SDSS for now.

Comment (DinoBektesevic) [reply trail/query/views.py]: `Metadata.objects.values('instrument').distinct()`

is the invocation. This returns a `QuerySet` (a list) of dicts of key instrument and instrument names I think. It should be a one or two line change tops.

**Pull request merged**